stdlib/strings/strings.go

```diff
@@ -382,7 +384,69 @@ func generateUnicodeIsFunction(name string, Fn func(rune) bool) values.Function
 	)
 }
 
+func generateStrLength(name string) values.Function {
+	return values.NewFunction(
+		name,
+		semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+			Parameters: map[string]semantic.PolyType{stringArgV: semantic.String},
+			Required:   semantic.LabelSet{stringArgV},
+			Return:     semantic.Int,
+		}),
+		func(args values.Object) (values.Value, error) {
+			v, ok := args.Get(stringArgV)
+			if !ok {
+				return nil, fmt.Errorf("missing argument %q", stringArgV)
+			}
+
+			if v.Type().Nature() == semantic.String {
+				arr := strings.Split(v.Str(), "")
+				l := len(arr)
+				return values.NewInt(int64(l)), nil
+			}
+
+			return nil, fmt.Errorf("procedure cannot be executed")
+		}, false,
+	)
+}
+
+func generateSubstring(name string) values.Function {
+	return values.NewFunction(
```

Comment:
As a small suggestion I would change these to be `var`s instead of functions as the function is only called once to produce a single value. 

```
var substring = values.NewFunction(...)
```

Then in the init function it becomes 
```

	flux.RegisterPackageValue("strings", "substring", substring)
```